internal/infrastructure/kubernetes/resource/resource.go

```diff
@@ -3,7 +3,7 @@
 // The full text of the Apache license is available in the LICENSE file at
 // the root of the repo.
 
-package utils
+package resource
```

Comment:
prefer `utils` than `resource` personally.

Reply:
IMO, `GetSelector` and `ExpectedServiceSpec` is the part of the infra resource